### PR TITLE
Handle a series of Exception for timestamp with timezone

### PR DESCRIPTION
**Pull request by JoJoBizarreAdventure**
Solving problems for issue https://github.com/trinodb/trino/issues/7502

### CODE REVIEW

File: plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/PartitionTable.java [hunk 1]
```diff
@@ -287,6 +287,9 @@ public static Object convert(Object value, Type type)
             // TODO the client sees the bytearray's tostring ouput instead of seeing actual bytes, needs to be fixed.
             return ((ByteBuffer) value).array();
         }
+        if (type instanceof Types.DateType) {
+            return Long.parseLong(value.toString());
```
Comment (JoJoBizarreAdventure) [inline plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/PartitionTable.java:291]: I have tried to add a branch for DateType, but it still goes to the branch for TimestampType.
What shall I do for this? @hashhar

File: core/trino-spi/src/main/java/io/trino/spi/predicate/NullableValue.java [hunk 2]
```diff
@@ -43,7 +45,9 @@ public NullableValue(Type type, Object value)
     {
         requireNonNull(type, "type is null");
         if (value != null && !Primitives.wrap(type.getJavaType()).isInstance(value)) {
-            throw new IllegalArgumentException(format("Object '%s' does not match type %s", value, type.getJavaType()));
+            if (!(type instanceof DateType && value instanceof LongTimestampWithTimeZone)) {
+                throw new IllegalArgumentException(format("Object '%s' does not match type %s", value, type.getJavaType()));
```
Comment (findepi) [inline core/trino-spi/src/main/java/io/trino/spi/predicate/NullableValue.java:49]: We shouldn't need to have this SPI change to make Iceberg connector work.
The code here was correct.

Comment (JoJoBizarreAdventure) [reply core/trino-spi/src/main/java/io/trino/spi/predicate/NullableValue.java:49]: But after casting it goes to the Timestamp branch, what shall I do for this? @findepi

**Comment by colebow** [comment]
:wave: @JoJoBizarreAdventure - this PR has become inactive. If you're still interested in working on it, please let us know, and we can try to get reviewers to help with that.

We're working on closing out old and inactive PRs, so if you're too busy or this has too many merge conflicts to be worth picking back up, we'll be making another pass to close it out in a few weeks.

**Comment by hashhar** [comment]
The issue this tries to fix has actually been addressed in https://github.com/trinodb/trino/pull/8647.

**Pull request closed without merging**